LibVLCSharp.Forms/Shared/Themes/Generic.xaml

```diff
@@ -0,0 +1,212 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<ResourceDictionary xmlns="http://xamarin.com/schemas/2014/forms" 
+                    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+                    xmlns:converters="clr-namespace:LibVLCSharp.Forms.Shared.Converters"
+                    xmlns:effects="clr-namespace:LibVLCSharp.Forms.Shared.Effects"
+                    x:Class="LibVLCSharp.Forms.Shared.Themes.Generic">
+    <converters:BufferingProgressToBoolConverter x:Key="BufferingProgressToBoolConverter" />
+    <converters:ObjectToBoolConverter x:Key="ObjectToBoolConverter" />
+
+    <Color x:Key="ForeColor">White</Color>
+    <Color x:Key="MainColor">#E57A00</Color>
+    <Color x:Key="ButtonColor">#A0E57A00</Color>
+
+    <OnPlatform x:Key="IconFontFamily" x:TypeArguments="x:String"
+                Android = "MaterialIcons-Regular.ttf#Material Icons"
```

Comment:
I didn't understand the discussion here. Does it need to be redistributed somehow by the app?

Reply:
Yes

Reply:
The default icons value are for the Material Icons web font. In my mind, the font should not be redistributed. If the developper wants to use the default icon values, he must include the Material Icons font in his project (we can't do it automatically in a nuget package anyway). Otherwise, he can override the buttons styles to set its own icons (image files or icons font values).

Reply:
Will the code points be the same for all iconic fonts? Can we replace it easily with another font? Wouldn't it be better to have XAML vectorial images resources instead of characters?

Reply:
> Will the code points be the same for all iconic fonts? Can we replace it easily with another font?

You should override all buttons style to set the text property with the correct code and set the IconFontFamily property.

> Wouldn't it be better to have XAML vectorial images resources instead of characters?

I would have preferred to use vectorial images (SVG or XAML vectorial) but it's complicated in Xamarin Forms, you need some third party libraries. The better solution would be to have our own icons font (but I'm not a graphic designer :smiley: ).

Reply:
Font Awesome could be an interesting option as well https://github.com/matthewrdev/fa2cs

We should probably use the platform's icons when possible and fallback on custom ones for the missing icons (cast and close captions and likely more).

Reply:
Or perhaps we can easily create an icon font from [these SVGs](https://code.videolan.org/videolan/vlc-android/tree/master/assets/images/svg).
I just tested with [IcoMoon app](https://icomoon.io/app/#/select), it seems easy to convert SVG to icons font. I need to test it with the MediaElement.

Reply:
I'm pretty sure that there are tools out there to convert svg to xaml. Isn't it possible to use <Path> in Xamarin.Forms? I didn't read which icons are required, but for play/pause, drawing rectangles/triangles should be enough

Reply:
You can't use a Path element in Xamarin.Form Xaml like in WPF. You must use a third party library like [SkiaSharp ](https://github.com/mono/SkiaSharp) to do an equivalent thing.
Icons font is the better way to display vectorial icons in Xamarin.Forms.

Reply:
I made a [custom icon font](https://github.com/kakone/LibVLCSharp.Forms.MediaPlayerElement/raw/master/Sample/Sample.Android/Assets/PlaybackControlsIcons.ttf) with the same identifiers than the Material Icons font (I used the [VLC Android SVGs](https://code.videolan.org/videolan/vlc-android/tree/master/assets/images/svg) and [IcoMoon app](https://icomoon.io/app/#/select) to generate the icon font).

So, we just need to replace the code by :
```
<OnPlatform x:Key="IconFontFamily" x:TypeArguments="x:String"
    Android = "PlaybackControlsIcons.ttf#PlaybackControlsIcons"
    iOS = "PlaybackControlsIcons" />
```

Reply:
Done. I'm not sure we want those icons on iOS though.

Reply:
Can't get this .ttf to work on iOS and I won't spend more hours trying to.

Additionally, expecting and using icons in the framework part (libvlcsharp.forms) that are _required_ to be included by each client part (end user apps) does not seem like a smooth experience to me, unless we automatically fallback on defaults system icons (or text if no default).

Reply:
VLC app icons are probably licensed GPLv2 (like the apps) so we probably cannot ship them with LibVLCSharp.Forms.

Reply:
Added the FontAwesome icon. The free pack seems open enough. And used this https://github.com/matthewrdev/fa2cs because the codes were giving me a headache.
Briefly looked into packaging those into the end user binary (apk, .app, whatever) from LibVLCSharp.Forms and:
- it doesn't look straightforward though still likely possible with a custom target
- it doesn't seem to be the convention for libraries to do that, though I thought it'd be a smoother dev experience (just like adding a WPF MediaElement and be done with it).
Thoughts?

Reply:
In my idea, it is not a smooth dev experience, but I thought to ask the dev to copy the font manually. Because if he wants to use another font or images, he doesn't need this FontAwesome.ttf file in his final package.

Reply:
Yes, but
- most people won't bother and will use the default icons (like the usual mediaElement)
- the font we provide could be disable with a csproj boolean property `<UseFontAwesome>false</UseFontAwesome>` in the user project.

Reply:
Think I found a more gracious way.
- Bundle fonts with `EmbeddedResource` with LibVLCSharp.Forms (if not told otherwise by user's csproj).
- Use XF platform effects to apply them.
Started work here https://code.videolan.org/mfkl/LibVLCSharp/commit/3058d14358c2f6009b7cf2ec23acd5f74d92da92 but will finish it in a next update.

Reply:
Turns out we will be able to use VLC iOS/Android icons after all...